doc/compiler.md

```diff
@@ -0,0 +1,205 @@
+# Using the ProtoDef compiler
+
+The ProtoDef compiler can convert your protocol JSON into javascript code that can read and write buffers directly instead of using the ProtoDef interpreter. Depending on the types, the expected speedups are in the range of x10 - x100.
+
+## Simple usage
+
+Let's take a simple ProtoDef definition and convert it to use the ProtoDef compiler:
+
+ProtoDef:
+```javascript
+const ProtoDef = require('protodef').ProtoDef
+
+// Create a ProtoDef instance
+const proto = new ProtoDef()
+proto.addTypes(require('./protocol.json'))
+
+// Encode and decode a message
+const buffer = proto.createPacketBuffer('mainType', result)
+const result = proto.parsePacketBuffer('mainType', buffer)
+```
+
+ProtoDef Compiler:
+```javascript
+const { ProtoDefCompiler } = require('protodef').Compiler
+
+// Create a ProtoDefCompiler instance
+const compiler = new ProtoDefCompiler()
+compiler.addTypesToCompile(require('./protocol.json'))
+
+// Compile a ProtoDef instance
+const compiledProto = await compiler.compileProtoDef()
```

Comment:
Its only because of closure-compiler if the option `{optimize: true}` is given to the compile function. I could make a compileProtoDefSync function that don't support optimization.

Reply:
it's probably ok to have it async, I was just wondering

Reply:
I added a sync version that will ease the integration.